### PR TITLE
feat(root): Implement Dropdown Component

### PR DESCRIPTION
**Pull request by mucahit**
### Description
- Dropdown component is done 🎉 
- Added few utility files that related to Dropdown

**To see the implementation [check it out](https://react-ui-toolkit.muco.now.sh/?path=/story/dropdown--dropdown-states)**

### CODE REVIEW

File: src/dropdown/list/DropdownList.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      ref={listRef}
+      data-testid={testid}
+      role={role}
+      aria-labelledby={ariaLabelledBy}
+      aria-hidden={ariaHidden}
+      aria-multiselectable={isMultiSelect}
+      className={containerClassName}>
+      {options.length ? (
+        options.map(function renderDropdownListItem(option, index) {
+          return (
```
Comment (jamcry) [inline src/dropdown/list/DropdownList.tsx:93]: I think we can convert this to arrow function to increase readability
```suggestion
        options.map((option, index) => (
```

File: src/dropdown/_dropdown.scss [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  @include truncate-line(calc(100% - #{$dropdown-header-horizontal-padding}));
+
+  flex: 1;
+
+  color: $dark-grey;
+
+  text-align: left;
+}
+
+svg {
```
Comment (edizcelik) [inline src/dropdown/_dropdown.scss:119]: Why are we modifying the svg tag styles here ?

File: src/dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+      {selectedOption && selectedOption.icon}
+
+      <span className={"dropdown-header-button-text"}>
+        {selectedOption ? selectedOption.title : deselectOptionTitle || placeholder}
+      </span>
+
+      {areOptionsFetching ? (
+        <Spinner spinnerColor={"black"} backgroundColor={"#EBEBEB"} />
+      ) : (
+        <CaretDownIcon aria-hidden={true} />
```
Comment (edizcelik) [inline src/dropdown/Dropdown.tsx:154]: Let's add a className to the svg icon

File: src/dropdown/list/item/DropdownListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import classNames from "classnames";
+
+export interface DropdownOption<Id = string, Context = any> {
+  id: Id;
+  customClassName?: string;
+  title: string;
+  CustomContent?: JSX.Element;
+  icon?: React.ReactNode;
+  subtitle?: string;
+  context?: Context;
```
Comment (edizcelik) [inline src/dropdown/list/item/DropdownListItem.tsx:13]: I think when typing the props, it is better to list required ones before the optional ones.
```suggestion
  id: Id;
  title: string;
  customClassName?: string;
  CustomContent?: JSX.Element;
  icon?: React.ReactNode;
  subtitle?: string;
  context?: Context;
```

File: src/dropdown/list/item/DropdownListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+export type TDropdownOptionSelectHandler<Id = string, Context = any> = (
+  option: DropdownOption<Id, Context> | null,
+  event?: React.SyntheticEvent<HTMLLIElement>
+) => void;
+
+export type TDropdownSelectedOption<Id = string, Context = any> =
```
Comment (edizcelik) [inline src/dropdown/list/item/DropdownListItem.tsx:21]: Let's remove `T` from type names

File: src/form/field/FormField.tsx [hunk 6]
```diff
@@ -37,13 +37,21 @@ function FormField(props: FormFieldProps) {
 
       {hasErrorMessage &&
         errorMessages?.map((message) => (
-          <FormFieldMessageRow type={"error"} message={message} />
+          <FormFieldMessageRow
+            key={`error.${message}`}
+            type={"error"}
+            message={message}
+          />
         ))}
```
Comment (edizcelik) [inline src/form/field/FormField.tsx:45]: I think it should be wrapped with `List` component. Same for helper messages below as well.

File: src/dropdown/_dropdown.scss (hunk 2 above)
Comment (mucahit) [reply src/dropdown/_dropdown.scss:119]: It was a mistake, fixed with latest prs

File: src/dropdown/Dropdown.tsx (hunk 3 above)
Comment (mucahit) [reply src/dropdown/Dropdown.tsx:154]: `react-svg-loader` doesn't accept `className` currently

**Pull request merged**